src/EnergyPlus/BaseboardRadiator.cc

```diff
@@ -569,8 +566,7 @@ namespace BaseboardRadiator {
 
         auto &baseboard = state.dataBaseboardRadiator;
 
-        if (baseboard->baseboards(BaseboardNum).ZonePtr <= 0)
-            baseboard->baseboards(BaseboardNum).ZonePtr = state.dataZoneEquip->ZoneEquipConfig(ControlledZoneNumSub).ActualZoneNum;
+        baseboard->baseboards(BaseboardNum).ZonePtr = ControlledZoneNum;
```

Comment:
Things like this could move to init. There is much improvement in these changes and much more to do.